whylog/config/rule.py

```diff
@@ -19,10 +19,10 @@ def serialize(self):
             "constraints": self._constraints,
         }
 
-    def get_new_parsers(self, old_parsers):
+    def get_new_parsers(self, parser_name_generator):
         new_parsers = []
         for parser in itertools.chain([self._effect], self._causes):
-            if parser.name not in old_parsers:
+            if parser_name_generator.is_free_parser_name(parser.name):
```

Comment:
Yes, if this condition equals False, we won't save parser. But this is case we want save rule's parsers, but some of then are reused, so they are in config base now.